content/800-guides/1200-prisma-postgres-realtime-on-cloudflare.mdx

```diff
@@ -0,0 +1,451 @@
+
+   ```terminal
+   npm create hono@latest realtime-app
+   ```
+
+2. Navigate into the newly created app directory:  
+
+   ```terminal
+   cd ./realtime-app
+   ```
+
+3. Follow the CLI prompts to configure your project:  
+   a. Select the `cloudflare-workers` template.  
+   b. Agree to install the project dependencies.  
+   c. Choose `npm` as your package manager (or use your preferred manager but update commands accordingly throughout this guide).  
+
+## 2. Set up Prisma in your application  
+
+
+1. Install Prisma CLI as a dev dependency:  
+
+   ```terminal
+   npm install prisma --save-dev
+   ```
+
+2. Install the [Prisma Accelerate client extension](https://www.npmjs.com/package/@prisma/extension-accelerate) as that's required for Prisma Postgres:  
+
+   ```terminal
+   npm i @prisma/extension-accelerate
+   ```
+
+2. Install the [Prisma Pulse client extension](https://www.npmjs.com/package/@prisma/extension-pulse) for real-time database updates:  
+
+   ```terminal
+   npm i @prisma/extension-pulse
+   ```
+
+3. Initialize Prisma in your application:  
+
+   ```terminal
+   npx prisma init
+   ```
+
+This will create:  
+- A `prisma` folder containing `schema.prisma`, where you will define your database schema.  
+- An `.env` file in the project root, which stores environment variables.  
+    :::note
+    You will not use `.env` files, as they are incompatible with Cloudflare Workers. You will delete this file later.
+    :::
+
+## 3. Create a Prisma Postgres instance  
+
+To store your app’s data, you’ll create a Prisma Postgres database instance using the Prisma Data Platform.
+
+1. Navigate to [Prisma Data Platform](https://console.prisma.io).  
+2. Click **New Project** and enter a name for your project in the **Name** field.  
+3. Under the **Prisma Postgres®** section, click **Get Started**.  
+4. Select a region close to your location from the **Region** dropdown.  
+5. Click **Create Project**. You’ll be redirected to the database setup page.  
+6. In the **Set up database access** section, copy the `DATABASE_URL` and `PULSE_API_KEY` environment variables. These will be required in the next steps.
+
+### 1. Configure development environment variables  
+
+1. In your project root, create a `.dev.vars` file to store environment variables:  
+
+   ```bash file=.dev.vars
+   DATABASE_URL=<your-database-url>
+   PULSE_API_KEY=<your-pulse-api-key>
+   ```
+
+2. Delete the `.env` file created by Prisma initialization, as `.env` is not compatible with Cloudflare Workers.  
+
+### 2. Update your Prisma schema  
+
+1. Open the `schema.prisma` file in the `prisma` folder.  
+2. Add the following model to define the structure of your database:  
+
+   ```prisma
+   generator client {
+     provider = "prisma-client-js"
+   }
+
+   datasource db {
+     provider = "postgresql"
+     url      = env("DATABASE_URL")
+   }
+
+   model Points {
+     id  Int @id @default(autoincrement())
+     x   Int
+     y   Int
+   }
+   ```
```

Comment:
the code snippet is indented with 3 spaces (it should be 2 or 4)

````suggestion
    ```prisma
    generator client {
      provider = "prisma-client-js"
    }

    datasource db {
      provider = "postgresql"
      url      = env("DATABASE_URL")
    }

    // add-start
    model Points {
      id  Int @id @default(autoincrement())
      x   Int
      y   Int
    }
    // add-end
    ```
````

Reply:
also, there are two spaces between `id` and `Int` and three between `x`/`y` and `Int`, please ensure the proper Prisma formatting 😄 

![image](https://github.com/user-attachments/assets/111c8369-30a0-4ea8-a63d-9770c81d4ca7)